spanner/cloud-client/backup_sample.py

```diff
@@ -0,0 +1,276 @@
+    ).format(database_id)
+    operations = instance.list_backup_operations(filter_=filter_)
+    for op in operations:
+        metadata = op.metadata
+        print("Backup {} on database {}: {}% complete.".format(
+            metadata.name, metadata.database,
+            metadata.progress.progress_percent))
+# [END spanner_list_backup_operations]
+
+
+# [START spanner_list_database_operations]
+def list_database_operations(instance_id):
+    spanner_client = spanner.Client()
+    instance = spanner_client.instance(instance_id)
+
+    # List the progress of restore.
+    filter_ = (
+        "(metadata.@type:type.googleapis.com/"
+        "google.spanner.admin.database.v1.OptimizeRestoredDatabaseMetadata)"
+    )
+    operations = instance.list_database_operations(filter_=filter_)
+    for op in operations:
+        print("Database {} restored from backup is {}% optimized.".format(
+            op.metadata.name, op.metadata.progress.progress_percent))
+# [END spanner_list_database_operations]
+
+
+# [START spanner_list_backups]
+def list_backups(instance_id, database_id, backup_id):
+    spanner_client = spanner.Client()
+    instance = spanner_client.instance(instance_id)
+
+    # List all backups.
+    print("All backups:")
+    for backup in instance.list_backups():
+        print(backup.name)
+
+    # List all backups that contain a name.
+    print("All backups with backup name containing \"{}\":".format(backup_id))
+    for backup in instance.list_backups(filter_="name:{}".format(backup_id)):
+        print(backup.name)
+
+    # List all backups for a database that contains a name.
+    print("All backups with database name containing \"{}\":".format(database_id))
+    for backup in instance.list_backups(filter_="database:{}".format(database_id)):
+        print(backup.name)
+
+    # List all backups that expire before a timestamp.
+    expire_time = datetime.utcnow().replace(microsecond=0) + timedelta(days=30)
+    print("All backups with expire_time before \"{}-{}-{}T{}:{}:{}Z\":".format(
+        *expire_time.timetuple()))
+    for backup in instance.list_backups(
+            filter_="expire_time < \"{}-{}-{}T{}:{}:{}Z\"".format(
+                *expire_time.timetuple())):
+        print(backup.name)
+
+    # List all backups with a size greater than some bytes.
+    print("All backups with backup size more than 100 bytes:")
+    for backup in instance.list_backups(filter_="size_bytes > 100"):
+        print(backup.name)
+
+    # List backups that were created after a timestamp that are also ready.
+    create_time = datetime.utcnow().replace(microsecond=0) - timedelta(days=1)
+    print("All backups created after \"{}-{}-{}T{}:{}:{}Z\" and are READY:".format(
+        *create_time.timetuple()))
+    for backup in instance.list_backups(
+            filter_="create_time >= \"{}-{}-{}T{}:{}:{}Z\" AND state:READY".format(
+                *create_time.timetuple())):
+        print(backup.name)
```

Comment:
One more thing. We have a pagination sample in the other languages. Was that every here? I may have missed that in the review.